pkg/controller/workspacerouting/solvers/oauth_proxy_container.go

```diff
@@ -0,0 +1,117 @@
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package solvers
+
+import (
+	"fmt"
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/common"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+)
+
+const proxyServiceAcctAnnotationKeyFmt string = "serviceaccounts.openshift.io/oauth-redirectreference.%s-%s"
+const proxyServiceAcctAnnotationValueFmt string = `{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"%s"}}`
+
+var openShiftProxySARFmt = `{"namespace": "%s", "resource": "pods", "name": "%s", "verb": "exec"}`
+
+func getProxyPodAdditions(proxyEndpoints map[string]proxyEndpoint, meta WorkspaceMetadata) *v1alpha1.PodAdditions {
+	var proxyContainers []corev1.Container
+	for _, proxyEndpoint := range proxyEndpoints {
+		proxyContainers = append(proxyContainers, getProxyContainerForEndpoint(proxyEndpoint, meta))
+	}
+	proxyVolumes := getProxyVolumes(proxyContainers)
+	serviceAcctAnnotations := getProxyServiceAcctAnnotations(proxyEndpoints, meta)
+
+	return &v1alpha1.PodAdditions{
+		Containers:                proxyContainers,
+		Volumes:                   proxyVolumes,
+		ServiceAccountAnnotations: serviceAcctAnnotations,
+	}
+}
+
+func getProxyContainerForEndpoint(proxyEndpoint proxyEndpoint, meta WorkspaceMetadata) corev1.Container {
+	proxyContainerName := fmt.Sprintf("%s-oauth-proxy-%s", meta.WorkspaceId, strconv.FormatInt(proxyEndpoint.upstreamEndpoint.Port, 10))
+
+	return corev1.Container{
+		Name: proxyContainerName,
+		Ports: []corev1.ContainerPort{
+			{
+				//Name:          endpoint.upstreamEndpoint.Name,
+				ContainerPort: int32(proxyEndpoint.publicEndpoint.Port),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		ImagePullPolicy: corev1.PullPolicy(config.ControllerCfg.GetSidecarPullPolicy()),
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "proxy-tls",
+				MountPath: "/etc/tls/private",
+			},
+		},
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+		Image:                    "openshift/oauth-proxy:latest",
+		Args: []string{
+			"--https-address=:" + strconv.FormatInt(proxyEndpoint.publicEndpoint.Port, 10),
+			"--http-address=127.0.0.1:" + strconv.FormatInt(proxyEndpoint.publicEndpointHttpPort, 10),
+			"--provider=openshift",
+			"--openshift-service-account=" + common.ServiceAccountName(meta.WorkspaceId),
+			"--upstream=http://localhost:" + strconv.FormatInt(proxyEndpoint.upstreamEndpoint.Port, 10),
+			"--tls-cert=/etc/tls/private/tls.crt",
+			"--tls-key=/etc/tls/private/tls.key",
+			"--cookie-secret=SECRET",
+			// Currently: block anyone who can't exec in the current namespace
+			"--openshift-sar=" + fmt.Sprintf(openShiftProxySARFmt, "", ""),
```

Comment:
```suggestion
```
`""` seems to be a default namespace but not the current. So, it does not work for me when I test workspace as a developer
![Screenshot_20200406_102950](https://user-images.githubusercontent.com/5887312/78533550-d144eb80-77f1-11ea-8971-457ffc3e06a1.png)

Reply:
If it's a namespace issue, then you should get a forbidden page; 500 internal server error is weird here.